xbmc/pvr/addons/PVRClient.cpp

```diff
@@ -405,6 +412,115 @@ bool CPVRClient::GetAddonProperties(void)
   try { strBackendHostname = m_pStruct->GetBackendHostname(); }
   catch (std::exception &e) { LogException(e, "GetBackendHostname()"); return false; }
 
+  /* timer types */
+  if (addonCapabilities.bSupportsTimers)
+  {
+    try
+    {
+      PVR_TIMER_TYPE types_array[PVR_ADDON_TIMERTYPE_ARRAY_SIZE];
+      int size = PVR_ADDON_TIMERTYPE_ARRAY_SIZE;
+
+      PVR_ERROR retval = m_pStruct->GetTimerTypes(types_array, &size);
+
+      if (retval == PVR_ERROR_NOT_IMPLEMENTED)
+      {
+        // begin compat section
+        CLog::Log(LOGWARNING, "%s - Addon %s does not support timer types. It will work, but not benefit from the timer features introduced with PVR Addon API 1.9.7.", __FUNCTION__, strFriendlyName.c_str());
+
+        // Create standard timer types (mostly) matching the timer functionality available in Isengard.
+        // This is for migration only and does not make changes to the addons obsolete. Addons should
+        // work and benefit from some UI changes (e.g. some of the timer settings dialog enhancements),
+        // but all old problems/bugs due to static attributes and values will remain the same as in
+        // Isengard. Also, new features (like epg search) are not available to addons automatically.
+        // This code can be removed once all addons actually support the respective PVR Addon API version.
```

Comment:
since we've bumped the minimum required api version because of the breaking changes, add-ons will not be backwards compatible. do we want this fluff in here?

we could add a header-only method in the pvr api easily to set backwards compatible entries if needed, but since all add-ons have to be updated anyway, shall we not require them to set the correct entries directly?